lib/elixir/test/elixir/kernel/cli_test.exs

```diff
@@ -96,10 +96,28 @@ defmodule Kernel.CLI.CompileTest do
   test "fails on missing patterns" do
     fixture = fixture_path "compile_sample.ex"
     output = elixirc('#{fixture} non_existing.ex -o #{tmp_path}')
-    assert :string.str(output, 'non_existing.ex') > 0, "expected non_existing.ex to be mentionned"
-    assert :string.str(output, 'compile_sample.ex') == 0, "expected compile_sample.ex to not be mentionned"
+    assert :string.str(output, 'non_existing.ex') > 0, "expected non_existing.ex to be mentioned"
+    assert :string.str(output, 'compile_sample.ex') == 0, "expected compile_sample.ex to not be mentioned"
     refute File.exists?(tmp_path("Elixir.CompileSample.beam")), "expected the sample to not be compiled"
   end
+
+  test "fails on missing write access to .beam file" do
```

Comment:
I experimented with moving to `Kernel.CLI.ParallelCompiler`, but it's less straightforward because of the `catch_exit` and `capture_io` stuff.  Actually, I couldn't even get it to work, not sure why.

And it would still require a side-effect during the test setup, because we need to create the beam file and set it to read-only before doing the compilation which we expect to fail.  So based on my attempts so far I'd prefer to leave it as is.